website/server/models/group.js

```diff
@@ -623,11 +642,20 @@ schema.methods.finishQuest = async function finishQuest (quest) {
     }
   });
 
-  let q = this._id === TAVERN_ID ? {} : {_id: {$in: this.getParticipatingQuestMembers()}};
+  let participants = this.getParticipatingQuestMembers();
```

Comment:
The problem is that participants depends on this.quest not being empty which happens on the following line and before the tavern logic. Because this.quest = {}; needs to be before the tavern logic and participants needs to be before that, I think this needs to remain above the tavern logic. Let me know if you have other thoughts here or if I'm misinterpreting something.